### PR TITLE
Triggers post

### PR DESCRIPTION
**Pull request by vtclose**

### CODE REVIEW

File: _posts/2023-06-28-triggers/index.md [hunk 1]
```diff
@@ -0,0 +1,185 @@
...
+
+## 1+1+1+1+…
+
+Now we’ve seen how we can match a single event, it’s time to scale it up to handle production workloads.
+
+Our goal is to be able to support millions of triggers running at the same time on event stream of tens of thousands events per second. This accounts for a fair amount of future growth so we don’t have to redesign a complex system in the next 5+ years.
+
+A naive solution to just check every event against every trigger condition doesn’t really scale all that well. Note that we instruct kafka to partition (shard) events by object ID - that doesn’t include organization ID, so even if you have multiple workers processing events, each of them has to be aware of every trigger at all times. With growing number of customers (and customers also growing in size), this approach breaks down pretty quickly - the memory overhead for each process is too big.
```
Comment (nsaje) [inline _posts/2023-06-28-triggers/index.md:139]: "kafka" -> "Kafka"

File: _posts/2023-06-28-triggers/index.md [hunk 2]
```diff
@@ -0,0 +1,185 @@
...
+- the condition doesn’t match regardless of related objects
+- the outcome depends on the data from related objects
+
+The first two outcomes are more or less trivial, but the third one means we have to go fetch the related object (or multiple!) from the primary datastore, extract the data from it in the same format suitable for matching against conditions, and then evaluate the remainder of the query again until we arrive at a definite yes or now.
+
+## 1+1+1+1+…
+
+Now we’ve seen how we can match a single event, it’s time to scale it up to handle production workloads.
+
+Our goal is to be able to support millions of triggers running at the same time on event stream of tens of thousands events per second. This accounts for a fair amount of future growth so we don’t have to redesign a complex system in the next 5+ years.
```
Comment (nsaje) [inline _posts/2023-06-28-triggers/index.md:137]: "on event stream" -> "on an event stream"

File: _posts/2023-06-28-triggers/index.md (hunk 1 above)
Comment (nsaje) [reply _posts/2023-06-28-triggers/index.md:139]: "with growing number" -> "with a growing number"

**Review by nsaje — COMMENTED** [review]
A couple of comments inline, mostly grammar related. I stopped adding grammar related comments though - I think we should just use some tool for that like Grammarly, WDYT? :)

**Review by AlecRosenbaum — APPROVED** [review]
🚀 awesome post!

**Review by neob91-close — APPROVED** [review]
Awesome!

**Pull request merged**